### ISSUE TITLE
[EF Core] 3.x breaking change: "Method not found: [...] IProperty.get_ClrType()"

### ISSUE THREAD

**Issue by zaneclaes** [dbroudy/LazyEntityGraph#21]
When attempting to use this repository with EF Core 3.1.4, I am getting the following:

```
System.MissingMethodException : Method not found: 'System.Type Microsoft.EntityFrameworkCore.Metadata.IProperty.get_ClrType()'.
   at LazyEntityGraph.EntityFrameworkCore.ModelMetadataGenerator.GetForeignKeyConstraint(INavigation navProp)
   at LazyEntityGraph.EntityFrameworkCore.ModelMetadataGenerator.GetConstraints(INavigation from, INavigation to)+MoveNext()
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at System.Linq.Set`1.UnionWith(IEnumerable`1 other)
   at System.Linq.Enumerable.DistinctIterator`1.FillSet()
   at System.Linq.Enumerable.DistinctIterator`1.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at LazyEntityGraph.Core.ModelMetadata..ctor(IEnumerable`1 entityTypes, IEnumerable`1 constraints)
   at LazyEntityGraph.EntityFrameworkCore.ModelMetadataGenerator.GenerateModelMetadata(DbContext ctx)
   at LazyEntityGraph.EntityFrameworkCore.ModelMetadataGenerator.LoadFromContext[TContext](Func`2 createContext)
```

Customization code:
```
LazyEntityGraphCustomization customization = new LazyEntityGraphCustomization(
        ModelMetadataGenerator.LoadFromContext<EfcTestDataContext>(options => new EfcTestDataContext(options)));
```

It [appears this method was removed in EF Core 3.0](https://stackoverflow.com/questions/59205665/upgraded-to-ef-core-3-1-and-now-iproperty-get-clrtype-is-missing).

**Comment by dbroudy** [comment]
It looks like it's still there in EF Core 3.1, but it's only in IPropertyBase, and when building against EF Core 2.0 the runtime doesn't figure this out.

I need to see if I can build two versions of LazyEntityGraph with different EF Core dependencies, or if that doesn't work, build the different targets (netstandard2.0 vs netstandard2.1) with different EF Core versions and publish that.

**Comment by dbroudy** [comment]
@zaneclaes I just released 2.3.1 that should resolve this issue. Let me know if you have any more trouble.

**Comment by ToniaDemchuk** [comment]
Have similar issue, but with EF core 6
```
System.MissingMethodException: Method not found: 'System.Type Microsoft.EntityFrameworkCore.Metadata.ITypeBase.get_ClrType()'..
  Stack Trace:
      at LazyEntityGraph.EntityFrameworkCore.ModelMetadataGenerator.<>c.<GenerateModelMetadata>b__1_0(IEntityType x)
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.HashSet`1.UnionWith(IEnumerable`1 other)
   at System.Collections.Generic.HashSet`1..ctor(IEnumerable`1 collection, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.DistinctIterator`1.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at LazyEntityGraph.Core.ModelMetadata..ctor(IEnumerable`1 entityTypes, IEnumerable`1 constraints)
   at LazyEntityGraph.EntityFrameworkCore.ModelMetadataGenerator.GenerateModelMetadata(DbContext ctx)
   at LazyEntityGraph.EntityFrameworkCore.ModelMetadataGenerator.LoadFromContext[TContext](Func`2 createContext)
```
